processing/src/main/java/io/druid/segment/FloatDimensionHandler.java

```diff
@@ -0,0 +1,518 @@
+    {
+      float startTime = System.currentTimeMillis();
+
+      valueWriter.close();
+
+      final ColumnDescriptor.Builder builder = ColumnDescriptor.builder()
+                                                               .setValueType(ValueType.FLOAT);
+
+      builder.addSerde(
+          FloatGenericColumnPartSerde.serializerBuilder()
+                                    .withByteOrder(IndexIO.BYTE_ORDER)
+                                    .withDelegate((FloatColumnSerializer) valueWriter)
+                                    .build());
+      final ColumnDescriptor serdeficator = builder.build();
+
+      makeColumn(v9Smoosher, dimensionName, serdeficator);
+
+    }
+
+    @Override
+    public boolean needsConvertMissingValues()
+    {
+      return true;
+    }
+
+    @Override
+    public boolean canSkip()
+    {
+      return false;
+    }
+
+    @Override
+    public Float getConvertedEncodedValue(Float value, int indexNumber)
+    {
+      return value;
+    }
+
+    @Override
+    public void setWriteV8()
+    {
+      useV8 = true;
+    }
+
+    @Override
+    public void closeWritersV8(
+        List<IntBuffer> rowNumConversions,
+        OutputSupplier<FileOutputStream> invertedOut,
+        OutputSupplier<FileOutputStream> spatialOut
+    ) throws IOException
+    {
+      valueWriterV8.close(
+          IndexIO.makeNumericDimFile(outDir, dimensionName, IndexIO.BYTE_ORDER)
+      );
+    }
+
+    public ArrayList<FloatBuffer> getDimConversions()
+    {
+      return null;
+    }
+
+    private void makeColumn(
+        final FileSmoosher v9Smoosher,
+        final String columnName,
+        final ColumnDescriptor serdeficator
+    ) throws IOException
+    {
+      final SerializerUtils serializerUtils = new SerializerUtils();
+
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      serializerUtils.writeString(baos, mapper.writeValueAsString(serdeficator));
+      byte[] specBytes = baos.toByteArray();
+
+      final SmooshedWriter channel = v9Smoosher.addWithSmooshedWriter(
+          columnName, serdeficator.numBytes() + specBytes.length
+      );
+      try {
+        channel.write(ByteBuffer.wrap(specBytes));
+        serdeficator.write(channel);
+      }
+      finally {
+        channel.close();
+      }
+    }
+  }
+
+  public class FloatDimensionColumnReader implements DimensionColumnReader<Float, Float>
+  {
+    private final Column column;
+    private final GenericColumn genericColumn;
+
+    public FloatDimensionColumnReader(Column column)
+    {
+      this.column = column;
+      this.genericColumn = column.getGenericColumn();
+    }
+
+    @Override
+    public ImmutableBitmap getBitmapIndex(
+        String value, BitmapFactory bitmapFactory, int numRows
+    )
+    {
+      if (column == null) {
+        return bitmapFactory.makeEmptyImmutableBitmap();
+      }
+      final float parsedFloat = Float.parseFloat(value);
```

Comment:
@fjy Added a try/catch for NumberFormatException, will throw a ParseException if that occurs